e107_plugins/pm/pm_shortcodes.php

```diff
@@ -575,13 +575,13 @@ public function sc_pm_delete($parm = '')
 			}
 
 
-			if($parm != '')
+			if(count($parm) == 0)
```

Comment:
yes if $parm was an empty string count($parm) would equal zero. If it is an empty array, it equals zero. But $parm !='' doesn't work for an array, only a string